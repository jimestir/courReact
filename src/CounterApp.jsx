import React, { useState } from 'react'
import { PropTypes } from 'prop-types'

function CounterApp ({ initialValue }) {
  const [value, setValue] = useState(initialValue)
  const increment = () => setValue(value + 1)
  const decrement = () => setValue(value - 1)
  const reset = () => setValue(0)
  return (
    <>
      <div>{value}</div>
      <div>
        <button onClick={increment}>+1</button>
        <button onClick={decrement}>-1</button>
        <button onClick={reset}>Reset</button>
      </div>
    </>
  )
}

export { CounterApp }
CounterApp.protoTypes = {
  initialValue: PropTypes.number.isRequiered
}
