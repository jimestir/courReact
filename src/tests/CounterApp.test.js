import { CounterApp } from '../CounterApp'
const { render, screen, fireEvent } = require('@testing-library/react')
// import { render, screen } from '@testing-library/react'
describe('Test in <CounterApp />', () => {
  const value = 10

  test('Debe hacer match con el snapshot', () => {
    const { container } = render(<CounterApp initialValue={value} />)
    expect(container).toMatchSnapshot()
  })

  test('Debe de mostrar el valor inicial de 100 <CounterApp initialValue={100}/>', () => {
    render(<CounterApp initialValue={100} />)
    expect(screen.getByText(100)).toBeTruthy()
  })

  test('Debe de decrementar con el boton -1', () => {
    render(<CounterApp initialValue={value} />)
    fireEvent.click(screen.getByText('-1'))
    // screen.debug()
    expect(screen.getByText(9)).toBeTruthy()
  })

  test('Debe hacer reset con el botton reset', () => {
    render(<CounterApp initialValue={value} />)
    fireEvent.click(screen.getByText('+1'))
    fireEvent.click(screen.getByText('+1'))
    fireEvent.click(screen.getByText('+1'))
    screen.debug()
    fireEvent.click(screen.getByRole('button', { name: 'button-reset' }))
    screen.debug()
    expect(screen.getByText(0)).toBeTruthy()
  })
})
