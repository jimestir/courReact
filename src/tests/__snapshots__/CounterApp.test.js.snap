// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <CounterApp /> Debe hacer match con el snapshot 1`] = `
<div>
  <div>
    10
  </div>
  <div>
    <button>
      +1
    </button>
    <button>
      -1
    </button>
    <button
      aria-label="button-reset"
    >
      Reset
    </button>
  </div>
</div>
`;
